File: sdkwork-backend-sdk-csharp/Api/ProjectApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class ProjectApi
    {
        private readonly HttpClient _client;

        public ProjectApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update an existing project
        /// </summary>
        public async Task<PlusApiResultPlusProjectVO?> UpdateAsync(PlusProjectForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusProjectVO>(ApiPaths.BackendPath("/project"), body);
        }

        /// <summary>
        /// Create a new project
        /// </summary>
        public async Task<PlusApiResultPlusProjectVO?> CreateAsync(PlusProjectForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusProjectVO>(ApiPaths.BackendPath("/project"), body);
        }

        /// <summary>
        /// Get projects by page
        /// </summary>
        public async Task<PlusApiResultPagePlusProjectVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusProjectVO>(ApiPaths.BackendPath("/project/list"), body, query);
        }

        /// <summary>
        /// Get all projects
        /// </summary>
        public async Task<PlusApiResultListPlusProjectVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusProjectVO>(ApiPaths.BackendPath("/project/list/all"), body);
        }

        /// <summary>
        /// Get a project by ID
        /// </summary>
        public async Task<PlusApiResultPlusProjectVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusProjectVO>(ApiPaths.BackendPath($"/project/{id}"));
        }

        /// <summary>
        /// Delete a project
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/project/{id}"));
        }
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/ProjectApi.swift
import Foundation

public class ProjectApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing project
    public func update(body: PlusProjectForm) async throws -> PlusApiResultPlusProjectVO? {
        let response = try await client.put(ApiPaths.backendPath("/project"), body: body)
        return response as? PlusApiResultPlusProjectVO
    }

    /// Create a new project
    public func create(body: PlusProjectForm) async throws -> PlusApiResultPlusProjectVO? {
        let response = try await client.post(ApiPaths.backendPath("/project"), body: body)
        return response as? PlusApiResultPlusProjectVO
    }

    /// Get projects by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusProjectVO? {
        let response = try await client.post(ApiPaths.backendPath("/project/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusProjectVO
    }

    /// Get all projects
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusProjectVO? {
        let response = try await client.post(ApiPaths.backendPath("/project/list/all"), body: body)
        return response as? PlusApiResultListPlusProjectVO
    }

    /// Get a project by ID
    public func getById(id: String) async throws -> PlusApiResultPlusProjectVO? {
        let response = try await client.get(ApiPaths.backendPath("/project/\(id)"))
        return response as? PlusApiResultPlusProjectVO
    }

    /// Delete a project
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/project/\(id)"))
        return response as? PlusApiResultBoolean
    }
}
